### ISSUE TITLE
Docker module crashes when trying to use 'volumes_from' parameter.

### ISSUE THREAD

**Issue by tdbishop** [ansible/ansible-modules-core#693]
#### Issue Type:

Bug Report
#### Ansible Version:

ansible 1.9 (devel 46b29c3f4b) 
#### Docker.py version:

0.7.1
#### Environment:

Ubuntu 14.10
#### Summary:

Docker module crashes when trying to use 'volumes_from' parameter.
#### Steps To Reproduce:

Playbook task file:

```
- name: Install postgres server Container
  docker:
    image=postgres
    volumes_from="dbdata"
    name=db
    restart_policy=always
```
#### Expected Results:

Docker creates container with volumes_from correctly set.
#### Actual Results:

Error:

```
Traceback (most recent call last):
  File "/home/feedback/.ansible/tmp/ansible-tmp-1422028300.96-15792328162405/docker", line 2543, in <module>
    main()
  File "/home/feedback/.ansible/tmp/ansible-tmp-1422028300.96-15792328162405/docker", line 917, in main
    containers = manager.create_containers(1)
  File "/home/feedback/.ansible/tmp/ansible-tmp-1422028300.96-15792328162405/docker", line 773, in create_containers
    containers = do_create(count, params)
  File "/home/feedback/.ansible/tmp/ansible-tmp-1422028300.96-15792328162405/docker", line 747, in do_create
    result = self.client.create_container(**params)
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 554, in create_container
    memswap_limit, cpuset, host_config
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 204, in _container_config
    raise errors.DockerException(message.format('volumes_from'))
docker.errors.DockerException: 'volumes_from' parameter has no effect on create_container(). It has been moved to start()
```

I'm not sure if this is an ansible issue or a docker-py issue. The docker-py code was changed from warning to throwing if volumes_from is included in create() https://github.com/docker/docker-py/blob/master/docker/client.py#L204 

Params in the ansible docker module are all stored on the DockerManager and passed into create_container() and start(), so perhaps they need to be filtered out of the create_container() call if version >= 1.10? I'm not sure what the correct fix is here.

create_container(): https://github.com/ansible/ansible-modules-core/blob/670098af2d5b3351382f82848fcc8fdb5744c8f8/cloud/docker/docker.py#L749 
start(): https://github.com/ansible/ansible-modules-core/blob/670098af2d5b3351382f82848fcc8fdb5744c8f8/cloud/docker/docker.py#L809

**Comment by SergeyZh** [comment]
I've created patch to fix this problem: https://github.com/ansible/ansible-modules-core/pull/730

**Comment by abadger** [comment]
This should be fixed in current devel and stable-1.9 branches.  Are you able to test those?

**Comment by abadger** [comment]
Closing as this should have been fixed when we merged #730 

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
